### PR TITLE
Credo

### PR DESCRIPTION
**Pull request by mememori**
Fix most credo concerns, remove ambiguous or useless checks, make the default linter the ci's one, add a config to check possible improvement oportunities

Note that the remaining line length prpblems are related to module vars, i might do another pr updating the check to ignore module vars' lengths

---
This change is [Reviewable](https://reviewable.io/reviews/hackerexperience/helix/23)

### CODE REVIEW

File: apps/hardware/lib/hardware/model/motherboard_slot.ex [hunk 1]
```diff
@@ -61,16 +60,19 @@ defmodule Helix.Hardware.Model.MotherboardSlot do
   def create_changeset(params) do
     %__MODULE__{}
     |> cast(params, @creation_fields)
-    |> validate_required([:motherboard_id, :link_component_type, :slot_internal_id])
     |> put_primary_key()
-    |> unique_constraint(:link_component_id, name: :motherboard_slots_link_component_id_index)
   end
```
Comment (mememori) [inline apps/hardware/lib/hardware/model/motherboard_slot.ex]: gotta add a |> changeset(params) here

**Comment by renatomassaro** [comment]
LGTM (I want custom "accept" comments!)

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/23#-:-KfkPSDrDOgunDN1F42_:b-uehvir)*

**Comment by mememori** [comment]
Review status: 16 of 18 files reviewed at latest revision, 5 unresolved discussions.

---

*[apps/hardware/lib/hardware/controller/motherboard_slot.ex, line 23 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/23#-KfkOBhZo0zTKWu4uvyJ:-KfkVjPB7LEedih5LcgC:b71wu3j) ([raw file](https://github.com/hackerexperience/helix/blob/54e93462dd95364829e1eac3eef9763f4118795e/apps/hardware/lib/hardware/controller/motherboard_slot.ex#L23)):*
<details><summary><i>Previously, renatomassaro (Renato Massaro) wrote…</i></summary><blockquote>

Why a single line for specs? The previous way (one line per return type) seems more readable
</blockquote></details>

As we discussed, it's lacking a guideline. I was thinking about going along with Elixir's stdlib default: one-liner for specs up to around 120 char, otherwise, left side on first line, rest on second-line (breaking on 80 char).

But we'll probably stick with your proposal: left side on a line, possible right side outcomes on a line each

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/23)*

**Comment by mememori** [comment]
Review status: 16 of 18 files reviewed at latest revision, 5 unresolved discussions.

---

*[apps/hardware/lib/hardware/model/network_connection.ex, line 59 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/23#-KfkMKfw0OTKF0JelfoE:-KfkWjYAm1ZxaNk6BiCc:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/54e93462dd95364829e1eac3eef9763f4118795e/apps/hardware/lib/hardware/model/network_connection.ex#L59)):*
<details><summary><i>Previously, renatomassaro (Renato Massaro) wrote…</i></summary><blockquote>

:+1: 
</blockquote></details>

Done.

---

*[apps/process/lib/process/controller/top.ex, line 279 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/23#-KfkMnW7GYyW_RwkAxV4:-KfkWlGTef7_LAenqmHn:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/54e93462dd95364829e1eac3eef9763f4118795e/apps/process/lib/process/controller/top.ex#L279)):*
<details><summary><i>Previously, renatomassaro (Renato Massaro) wrote…</i></summary><blockquote>

much better :wolf: 
</blockquote></details>

Done.

---

*[apps/process/lib/process/controller/top/allocator/plan.ex, line 72 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/23#-KfkN2GjpItRVXd9gXZM:-KfkWlpEyy2hrqE2Swft:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/54e93462dd95364829e1eac3eef9763f4118795e/apps/process/lib/process/controller/top/allocator/plan.ex#L72)):*
<details><summary><i>Previously, renatomassaro (Renato Massaro) wrote…</i></summary><blockquote>

1000x better. that's why it's preferable to bind the functional magic into a meaningful variable name
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/23)*

**Comment by renatomassaro** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/23)*

**Pull request merged**